From f166de260efaa396749fcfebf64208caa9fc1e8d Mon Sep 17 00:00:00 2001
From: Mark Harfouche <mark.harfouche@gmail.com>
Date: Thu, 28 Aug 2025 17:19:00 -0400
Subject: [PATCH] don't use delete with malloc on store

---
 src/core/common/ojph_mem.h | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/src/core/common/ojph_mem.h b/src/core/common/ojph_mem.h
index f14c6c3..c8166ca 100644
--- a/src/core/common/ojph_mem.h
+++ b/src/core/common/ojph_mem.h
@@ -81,8 +81,7 @@ namespace ojph {
       {
         // We should be here once only, because, in subsequent, calls we
         // should have size_data + size_obj <= allocated_data
-        if (store)
-          delete[] store;
+        free(store);
         allocated_data = size_data + size_obj;
         allocated_data = allocated_data + (allocated_data + 19) / 20; // 5%
         store = malloc(allocated_data);
